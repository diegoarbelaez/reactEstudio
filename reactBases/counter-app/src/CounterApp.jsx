import { useState } from "react";
import PropTypes from "prop-types";

export const CounterApp = ({ value }) => {
  const [counter, setCounter] = useState(value);
  const [nombre, setNombre] = useState('Diego')

  const handleAdd = () => {
    setCounter(counter + 1);
  };

  const handleRes = () =>{
    setCounter(counter -1 );
  }

  const handleReset = () => {
    setCounter(value)
  }

  
  


  return (
    <>
      <div>CounterApp</div>
      <div>{counter}</div>
      
      <button onClick={handleAdd}>+1</button>
      <button onClick={handleRes}>-1</button>
      <button onClick={handleReset}>Reset</button>
      

    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
};
