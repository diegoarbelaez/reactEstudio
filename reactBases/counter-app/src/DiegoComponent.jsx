import PropTypes from 'prop-types';


export const DiegoComponent = ({nombre, apellido}) => {
  return (
    <div>{nombre} {apellido} Componente!</div>
  )
}


DiegoComponent.propTypes = {
  nombre: PropTypes.string.isRequired,
  apellido: PropTypes.string.isRequired
}


DiegoComponent.defaultProps = {
  nombre: 'Angelica',
  apellido: 'Ceballos',
  edad: 30
}