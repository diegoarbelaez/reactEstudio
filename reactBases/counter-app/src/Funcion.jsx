
const cuenta = {
    saldo: 500,
    titular: 'Diego Arbelaez'
}


function getSaldo ( cuenta ){
    return cuenta.saldo;
}

function getTitular ( cuenta){
    return cuenta.titular
}


export const Funcion = () => {
  return (
    <>
    <div>Obtener Saldo Cuenta</div>
    <div>Saldo: {getSaldo(cuenta)}</div>
    <div>Titular: {getTitular(cuenta)}</div>

    </>
  )
}
