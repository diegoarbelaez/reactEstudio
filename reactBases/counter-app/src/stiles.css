html, body {
    background-color: aquamarine;
    color: black;
    font-family: 'Courier New', Courier, monospace;
    font-size: 40px;
}

button{
    font-size: 15px;
    padding: 10px;
    margin: 3px;
}