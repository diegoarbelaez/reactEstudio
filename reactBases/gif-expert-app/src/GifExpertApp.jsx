import { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

export const GifExpertApp = () => {
  const [categories, setCategories] = useState([
    "Saint Seiya"
  ]);

  const onNewCategory = (value) => {
    // setCategories([value, ...categories])
    if (categories.includes(value)) {
      return;
    }
    setCategories([...categories, value]); // Agrega al final del arreglo
  };

  return (
    <>
      <h1>GifExpertApp</h1>
      <AddCategory onNewCategory={onNewCategory} valor="Diego" />
      {categories.map((element) => (
        <GifGrid key={element} category={element} />
      ))}
    </>
  );
};
