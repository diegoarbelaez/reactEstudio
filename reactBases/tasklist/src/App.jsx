import React from 'react'
import TaskForm from './components/TaskForm';
import { TaskList } from './components/TaskList';
import { tareas } from './assets/tasks'
import { useState, useEffect } from 'react';


const App = () => {

  const [tasks, setTasks] = useState([]);

  //Very first time load the tasks, just one time
  useEffect(() => {
    setTasks(tareas);
  }, [])

  function addTask(tituloTarea, descripcionTarea) {
    setTasks([...tasks, {
      title: tituloTarea,
      id: tasks.length,
      description: descripcionTarea
    }])
    console.log(tasks)
  }

  function deleteTask(idTarea) {
    setTasks(tasks.filter(element => element.id != idTarea));
  }


  return (
    <main className='bg-zinc-900 h-screen'>
      <div className='container mx-auto'>
        <TaskForm agregarTarea={addTask} />
        <TaskList tareas={tasks} deleteTask={deleteTask} />
      </div>
    </main>
  )
}
export default App;



