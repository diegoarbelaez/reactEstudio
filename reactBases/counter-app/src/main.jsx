import React from "react";
import ReactDOM from "react-dom/client";
import { CounterApp } from "./CounterApp";
import { DiegoComponent } from "./DiegoComponent";
import { Funcion } from "./Funcion";
import {HelloWorldApp} from './HelloWorldApp';

import './stiles.css';


ReactDOM.createRoot(document.getElementById("root")).render(
  <React.StrictMode>
    {/* <HelloWorldApp />
    <DiegoComponent nombre='Diego!' apellido='Arbelaez' edad={41} />
    <Funcion /> 
    <CounterApp value={20} />
    <DiegoComponent nombre='Diego' apellido='Arbelaez' edad={41} />*/ 
    }
     <CounterApp value={20} />
   
  </React.StrictMode>
);
