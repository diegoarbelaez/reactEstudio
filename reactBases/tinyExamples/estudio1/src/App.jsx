import { useState } from 'react'
import './App.css'
import { BoxColor } from './components/BoxColor'
import { ConfigColor } from './components/ConfigColor'

function App() {
  const [count, setCount] = useState(0)

  return (
    <>
      
      <ConfigColor />
    </>
  )
}

export default App
