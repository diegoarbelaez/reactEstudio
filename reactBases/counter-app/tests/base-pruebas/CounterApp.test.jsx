import {fireEvent, render, screen} from '@testing-library/react';
import { CounterApp } from '../../src/CounterApp';


describe('Prueba Counter App', ()=>{


    test('Match con el Snapshot', ()=>{
        const { container } = render (<CounterApp value={20} />);
        expect (container) .toMatchSnapshot();
    })

    test('Debe Mostrar 100 al inicio', ()=>{
        render(<CounterApp value={100} />);
        const valor = screen.getByText(100);
        expect (valor.innerHTML) .toContain("100");
    })


    test('Debe incrementar', ()=>{
        render(<CounterApp value={100} />);
        fireEvent.click( screen.getByTestId('boton-incrementar'));
        fireEvent.click( screen.getByTestId('boton-incrementar'));
        fireEvent.click( screen.getByTestId('boton-incrementar'));
        expect ( screen.getByTestId('div-contador').innerHTML).toContain("103")
        screen.debug();
    })


})