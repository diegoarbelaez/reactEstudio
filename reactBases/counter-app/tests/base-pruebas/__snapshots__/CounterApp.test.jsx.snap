// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba Counter App Match con el Snapshot 1`] = `
<div>
  <div>
    CounterApp
  </div>
  <div
    data-testid="div-contador"
  >
    20
  </div>
  <button
    data-testid="boton-incrementar"
  >
    +1
  </button>
  <button>
    -1
  </button>
  <button>
    Reset
  </button>
</div>
`;
